### PR TITLE
fix: use the appropriate ID when trying to import `github_team_members` objects

### LINKED ISSUE #608: Error: Unexpected ID format (""), expected numerical ID. strconv.ParseInt: parsing "": invalid syntax
_This issue was originally opened by @pbharambe as hashicorp/terraform#26923. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._
    
<hr>
    

Getting "strconv.ParseInt: parsing" error while destroying exiting team membership in terraform apply.

### Terraform Version
```
Terraform v0.12.24
```

### Terraform Configuration Files
```terraform
repositories.tf

variable "github_token" {}

provider "github" {
  version = "~> 2.2.1"

  base_url     = "https://api.github.com/"
  organization = "pankajlab"
  token        = "${var.github_token}"
}

/***********************************************************************************************************************
 * TEAMS
 */
resource "github_team" "app-developers" {
  name        = "app-developers"
  description = "All app Developers."
  privacy     = "closed"
}

resource "github_team" "admin" {
  name    = "admin"
  privacy = "closed"
}

resource "github_team_membership" "pbharambe" {
  team_id  = "${github_team.admin.id}"
  username = "pbharambe"
  role     = "maintainer"
}

locals {
  app_team_ids = [
    "${github_team.app-developers.id}"
  ]
}

/***********************************************************************************************************************
 * REPOSITORIES
 */

module "cf-environments" {
  source = "./modules/github"

  repository_name = "cf-environments"
  description     = "Configurations defining Cloud Foundry environments along with the tools for managing them."

  branches_to_protect = ["master"]

  admin_team_id = "${github_team.admin.id}"
  team_ids      = "${local.app_team_ids}"
}

module "database-management" {
  source = "./modules/github"

  repository_name = "database-management"

  branches_to_protect = ["master"]

  admin_team_id = "${github_team.admin.id}"
  team_ids      = "${local.app_team_ids}"
}
------------------------------------------------------------------------------------
main.tf

variable "repository_name" {}

variable "auto_init" {
  default = true
}

variable "description" {
  default = ""
}

variable "private" {
  default = false
}

variable "has_downloads" {
  default = true
}

variable "has_issues" {
  default = true
}

variable "has_wiki" {
  default = true
}

variable "allow_merge_commit" {
  default = false
}

variable "allow_squash_merge" {
  default = true
}

variable "allow_rebase_merge" {
  default = false
}

variable "branches_to_protect" {
  type    = "list"
  default = ["master"]
}

variable "admin_team_id" {
  default = ""
}

variable "team_ids" {
  type    = "list"
  default = []
}

variable "enforce_admins" {
  default = true
}

resource "github_repository" "main" {
  name        = "${var.repository_name}"
  description = "${var.description}"

  private = "${var.private}"

  has_downloads = "${var.has_downloads}"
  has_issues    = "${var.has_issues}"
  has_wiki      = "${var.has_wiki}"

  allow_merge_commit = "${var.allow_merge_commit}"
  allow_squash_merge = "${var.allow_squash_merge}"
  allow_rebase_merge = "${var.allow_rebase_merge}"

  auto_init = "${var.auto_init}"

  lifecycle {
    prevent_destroy = true
  }
}

resource "github_team_repository" "admin" {
  team_id    = "${var.admin_team_id}"
  repository = "${github_repository.main.name}"
  permission = "admin"
}

resource "github_team_repository" "main" {
  count = "${length(var.team_ids)}"

  team_id    = "${var.team_ids[count.index]}"
  repository = "${github_repository.main.name}"
  permission = "push"
}

resource "github_branch_protection" "main" {
  count = "${length(var.branches_to_protect)}"

  repository = "${github_repository.main.name}"
  branch     = "${var.branches_to_protect[count.index]}"

  required_pull_request_reviews {
    dismiss_stale_reviews = true
    dismissal_teams       = ["${var.admin_team_id}"]
  }

  enforce_admins = false
}



```

### Expected Behavior
The membership should have removed first and then the user from the organization.

### Actual Behavior
```
github_team_membership.pbharambe: Destroying... [id=4275642:pbharambe]

Error: Unexpected ID format (""), expected numerical ID. strconv.ParseInt: parsing "": invalid syntax
```

### Steps to Reproduce
1. `terraform init`
2. `terraform plan`
3. `terraform apply`

### PR DESCRIPTION
**Pull request by bison-brandon**
When importing a `github_team_member`, the import command passes in the data as the
"Id", not the "team_id", causing the import step to fail when parsing an empty string
as a number. Should fix #608, as I hit the same error.

### CODE REVIEW

**Comment by bison-brandon** [comment]
I tested this by following [these notes](https://github.com/integrations/terraform-provider-github/tree/master/CONTRIBUTING.md#building-the-provider) and running `terraform import github_team_members.team_members <ID>`. I do not know how to unit test this, since it seems like the current tests ought to have caught it.

**Comment by SpencerMalone** [comment]
Good catch! I actually had also just dug in and realized this was the problem. The generic error actually had me all tripped up in thinking it was the schema validation at first. I think that old issue from 2020 is something else, but this definitely solves a very active problem in the recent release.

My only Q (and this comes from a place of ignorance w/ the codebase):

Do we need to repeat the pattern from `resource_github_team_membership.go` of...
```
	// We intentionally set these early to allow reconciliation
	// from an upstream bug which emptied team_id in state
	// See https://github.com/integrations/terraform-provider-github/issues/323
	d.Set("team_id", teamIdString)
	d.Set("username", username)
```
@ https://github.com/integrations/terraform-provider-github/blob/main/github/resource_github_team_membership.go#L85-L99 ?

**Comment by bison-brandon** [comment]
I believe line 85 using the split on `d.Id()` makes it safe in this case, since it seems like the _id_ in `import x <id>` is set as the value of `d.Id()` -- I haven't tested that import case, so I cannot be sure.

**Comment by SpencerMalone** [comment]
@kfcampbell any chance this could get bumped up in priority? It effectively makes the newest resource added unusable until this is merged. If there's another individual or group you'd like tagged, lemme know!

**Comment by bison-brandon** [comment]
For anyone blocked by this, you can take the patch, follow [these steps](https://github.com/integrations/terraform-provider-github/blob/main/CONTRIBUTING.md#building-the-provider), and run the imports. Since this only breaks imports, the rest of your workflow can remain the same after your import step.

**Review by kfcampbell — APPROVED** [review]
Thank you for the contribution! I'll get this released shortly.

**Pull request merged**
